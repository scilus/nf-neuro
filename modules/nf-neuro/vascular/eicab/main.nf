
process VASCULAR_EICAB {
    tag "$meta.id"
    label 'process_medium'

    container "${ 'felixdumais1/eicab:v1.0.1' }"
    containerOptions {
        (workflow.containerEngine == 'docker') ? '--entrypoint ""' : ''
    }

    input:
        tuple val(meta), path(in_tof)

    output:
        tuple val(meta), path("*_eicab")                  , emit: eicabdirectory
        tuple val(meta), path("*_eicab/*_eICAB_CW.nii.gz"), emit: eicabcw
        path "versions.yml"                               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def out_folder = task.ext.out_folder ?: "${prefix}_eicab"
    def resolution = task.ext.resolution ?: "0.625"
    """
    /vessel_segmentation_snaillab/eICAB.sh -t $in_tof \
                                           -o $out_folder \
                                           -r $resolution

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vessel-segmentation: \$(pip list | grep vessel-segmentation | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir -p ${prefix}_eicab/

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vessel-segmentation: \$(pip list | grep vessel-segmentation | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
