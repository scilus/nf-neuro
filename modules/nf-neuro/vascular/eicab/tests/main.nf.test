nextflow_process {

    name "Test Process VASCULAR_EICAB"
    script "../main.nf"
    process "VASCULAR_EICAB"

    tag "modules"
    tag "modules_nfcore"
    tag "vascular"
    tag "vascular/eicab"

    test("segmentation - eicab") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    [ file("/workspaces/nf-neuro/data/sub-OAS30001_ses-d2430_acq-TOF_angio_resampled.nii.gz") ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.eicabdirectory.get(0).get(1)).name,
                    file(process.out.eicabcw.get(0).get(1)).name,
                    process.out.versions).match()}
            )
        }
    }
}
