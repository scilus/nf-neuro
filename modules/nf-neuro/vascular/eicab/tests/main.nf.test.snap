{
    "segmentation - eicab": {
        "content": [
            "test_eicab",
            "sub-OAS30001_ses-d2430_acq-TOF_angio_resampled_eICAB_CW.nii.gz",
            [
                "versions.yml:md5,6b433c9a3e91fd2a59ac92510ccee4dd"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-12T02:51:01.189807007"
    }
}